cpp/ql/src/Security/CWE/CWE-611/XXE.ql

```diff
@@ -163,6 +163,13 @@ class ParseFunction extends Function {
   ParseFunction() { this.getClassAndName("parse") instanceof AbstractDOMParserClass }
 }
 
+/**
+ * The `createLSParser` function that returns a newly created `LSParser` object.
+ */
+class CreateLSParser extends Function {
+  CreateLSParser() { this.hasName("createLSParser") }
```

Comment:
Is this name too general to just be captured by the function name? If I were to create a function that constructed a class that captured, say, the output of `ls` from the command line I'd call this function `createLSParser` (hm, but maybe with a different casing, though). Should we check either that the return type is of the right type, or that the function lives in some namespace?

Reply:
Yeah, I should probably figure out the namespace this should be in and check that as well.

Reply:
It turns out the namespace name varies depending on the Xerces version, and its a class method but I'm not entirely confident the class name is stable.  So I've added matching on the return type, which the docs are clear is `DOMLSParser *`.  That should rule out any false matches.

Reply:
Fantastic!